File: thingseong/두 수의 합.py
import sys

n = int(sys.stdin.readline())
A = list(map(int, sys.stdin.readline().split()))
x = int(sys.stdin.readline())
A = sorted(A)
cnt = 0

l, r = 0, n-1

while(l < r):
    if A[l] + A[r] == x:
        cnt += 1
        l = l + 1
    elif A[l] + A[r] < x:
        l = l + 1
    else:
        r = r - 1

print(cnt)

File: vgihan/baekjoon/두수의합.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let n;
let inputs;

rl.on("line", function (line) {
  if (!n) {
    n = parseInt(line);
    return;
  }
  if (!inputs) {
    inputs = line.split(" ").map((v) => parseInt(v));
    return;
  }
  console.log(solution(parseInt(line), inputs));
  rl.close();
}).on("close", function () {
  process.exit();
});

function solution(x, inputs) {
  const arr = inputs.sort((a, b) => a - b);
  let i = 0;
  let j = arr.length - 1;
  let count = 0;

  while (i < j) {
    const sum = arr[i] + arr[j];
    if (sum > x) {
      j--;
      continue;
    }
    if (sum < x) {
      i++;
      continue;
    }
    count++;
    i++;
    j--;
  }

  return count;
}
